[TS-245] Add autocomplete stock info API endpoint

bowe99 opened the pull request:
http://localhost:5000/api/v1/searchRecommendations/{searchString}

![image](https://user-images.githubusercontent.com/50396173/148663735-f9f7191f-113c-48d5-a9f4-9b3c6ed45627.png)

bowe99 commented:
I still need to write some tests but the code works for what I need in the frontend

tungleu commented on src/controllers/search.controller.ts:
```diff
@@ -0,0 +1,33 @@
+		}
+		catch (error: any) {
+            if(error.response == undefined) {
+                res.send(messages.internalError(error.message));
```
nit: add return here

tungleu commented on src/controllers/search.controller.ts:
```diff
@@ -0,0 +1,33 @@
+            if(error.response == undefined) {
+                res.send(messages.internalError(error.message));
+            } else {
+                res.send(messages.internalError(error.response.data.message))
```
same here

tungleu reviewed (dismissed):
Just a few nit, awesome work! :)

kennguyen0303 commented on src/controllers/search.controller.ts:
```diff
@@ -0,0 +1,33 @@
+            if(process.env.ALPHAVANTAGE_API_KEY == undefined) {
+                throw new Error("Alphavantage api key is undefined");
+            }
+			const response = await axios.get(`${url}&keywords=${searchString}&apikey=${process.env.ALPHAVANTAGE_API_KEY}`, {
+				headers: { 'User-Agent': 'request' },
+			});
+			searchResults = response.data;
+			return res.send(messages.successMessage("success", "searchResult", searchResults))
```
```suggestion
			const {data}= await axios.get(`${url}&keywords=${searchString}&apikey=${process.env.ALPHAVANTAGE_API_KEY}`, {
				headers: { 'User-Agent': 'request' },
			});
			return res.send(messages.successMessage("success", "searchResult", data))
```

We can shorten the code this way, avoid creating unnecessary variable. Replace `response` by `{data}`

kennguyen0303 reviewed (changes requested):
Great work! I was surprised since you made it without any instruction. I think Damien comment on the `return` is necessary. Although it won't make any technical issue here, it will follow our convention to finish a function. 

For future development, I sometimes forgot the `return res.send(..)` and only write `res.send(..)`. Because a function will not stop unless it reads all lines or reach a `return` statement, if you have two `res.send(..)` in a function and somehow the function manages to run through both of them,  the front-end will only receive the first `res.send(..)` and an error is triggered in the back-end saying `A response has already been sent out, another response cannot be sent`

kennguyen0303 commented on src/routes/v1/search.route.ts:
```diff
@@ -0,0 +1,17 @@
+
+	// All paths have the prefix /api/v1/session/
+
+	router.get("/:searchString", (req, res) => {
+		accountController.getStocksSuggestions(req, res);
+	});
```
```suggestion
	router.get("/:searchString", async (req, res) => {
		await accountController.getStocksSuggestions(req, res);
	});
```

Adding `async/await` here for two purposes:
1. Coding convention to show it is an asynchronous function
2. Avoid unexpected behavior

kennguyen0303 commented:
hey @bowe99 , thanks for addressing the suggestions. I asked Damien to make a service and add unit test for his part. It's not gonna take long to change. You have mostly everything there, just need to wrap the `axios call into a method (inside a new class maybe call it `StockSearchService`). 

You can check out his changes here: https://github.com/TradeShare490/TradeShare-BE/pull/25/files

bowe99 commented:
> hey @bowe99 , thanks for addressing the suggestions. I asked Damien to make a service and add unit test for his part. It's not gonna take long to change. You have mostly everything there, just need to wrap the `axios call into a method (inside a new class maybe call it `StockSearchService`).
> 
> You can check out his changes here: [9bda9b4](https://github.com/TradeShare490/TradeShare-BE/commit/9bda9b4214463379473a453022be189d9d0d7640)

Can you approve and I'll do it in a separate sub-task? just so I can start working on the frontend part and get it merged

tungleu reviewed (dismissed):
nice work on making it as a service and adding test, I assume the test failed since @kennguyen0303 needs to add the environment variable to repo's secret for CI to run?

bowe99 commented:
> nice work on making it as a service and adding test, I assume the test failed since @kennguyen0303 needs to add the environment variable to repo's secret for CI to run?

Yeah exactly he should be adding it now

kennguyen0303 reviewed (approved):
Working now, lgtm ! Good job :+1:
